cpp/dolfinx/fem/discreteoperators.h

```diff
@@ -123,4 +123,199 @@ void assemble_discrete_gradient(const fem::FunctionSpace& V0,
     mat_set(dofmap1->cell_dofs(c), dofmap0->cell_dofs(c), Ae);
   }
 }
+
+/// @brief Assemble an interpolation operator matrix
+///
+/// This function creates a discrete operator \f$A\f$ that interpolates
+/// a function \f$V_0\f$ into a compatible space \f$V_1\f$, i.e.
```

Comment:
What is a 'compatible space'?

Reply:
A space with the correct value shape. I.e. you cannot interpolate a scalar function into a vector space or opposite.